sdk/src/Core/Amazon.Runtime/SharedInterfaces/ICoreAmazonSTS.cs

```diff
@@ -34,18 +32,26 @@ public interface ICoreAmazonSTS
         /// <returns></returns>
         AssumeRoleImmutableCredentials CredentialsFromAssumeRoleAuthentication(string roleArn, string roleSessionName, AssumeRoleAWSCredentialsOptions options);
 
-#if !BCL // In the NETSTANDARD flavors of the SDK ICoreAmazonSTS is declared without CredentialsFromSAMLAuthentication,
-    }             // we cannot add a new method to the interface for backward compatibility concerns. 
+        /// <summary>
+        /// <para>
+        /// This method is used internally to access the Amazon Security Token
+        /// service within other service assemblies.
+        /// Please use AmazonSecurityTokenServiceClient to access the Amazon Security Token
+        /// service instead.
+        /// </para>
+        /// Use Amazon Security Token Service to assume a role.
+        /// <remarks>
+        /// Proxy settings that are required for the HTTPS and STS calls made during the authentication/credential
+        /// generation process are supported and should have been configured on the STS ClientConfig instance
+        /// associated with the STS client instance exposing this interface.
+        /// </remarks>
+        /// </summary>
+        /// <param name="roleArn">The Amazon Resource Name (ARN) of the role to assume.</param>
+        /// <param name="roleSessionName"> An identifier for the assumed role session.</param>
+        /// <param name="options">Options to be used in the call to AssumeRole.</param>
+        /// <returns></returns>
+        Task<AssumeRoleImmutableCredentials> CredentialsFromAssumeRoleAuthenticationAsync(string roleArn, string roleSessionName, AssumeRoleAWSCredentialsOptions options);
 
-    /// <summary>
-    /// ICoreAmazonSTS_SAML is not meant to be used directly. It defines Security Token
-    /// service with basic .NET types and allows other services to be able to use the service as 
-    /// a runtime dependency. This interface is implemented by the AmazonSecurityTokenServiceClient 
-    /// defined in the AWSSDK.SecurityToken assembly.
-    /// </summary>
-    public interface ICoreAmazonSTS_SAML
```

Comment:
Question: Why did you delete `ICoreAmazonSTS_SAML` but not the `ICoreAmazonSTS_WebIdentity` interface? Would consolidating all the methods in a single interface be an option since this is targeting V4?

Reply:
Fixed in latest commit https://github.com/aws/aws-sdk-net/pull/3681/commits/e9ede6c25ed4a3e7e2ff9ea18b0ce7ccefcf7750. Will initiate a dry run.